### DISCUSSION TITLE
不同的消费者订阅同一个topic的不同tag，消息多的tag是否会影响消息少tag的消费速度

### DISCUSSION THREAD

**Discussion by oswardlx** [apache/rocketmq#8005] [General]
不同的消费者订阅同一个topic的不同tag，消息多的tag是否会影响消息少tag的消费速度

**Comment by biningo** [comment]
同一个消费者组的话，里面的消费者的订阅关系必须是一致的，文档：https://rocketmq.apache.org/docs/bestPractice/05subscribe

你指的是不同的消费者组订阅不同的tag是否会互相影响？这必然不会，每个消费者组都相当于是独立的

**Reply by good-134** [reply]
这是订阅关系不一致了吧。 同一 ConsumerGroup 下的三个Consumer实例C1、C2和C3分别都订阅了TopicA，但是C1订阅TopicA的Tag为Tag1，C2和C3订阅的TopicA的Tag为Tag2，订阅同一Topic的Tag不一致，不符合订阅关系一致性原则

**Reply by biningo** [reply]
对的，订阅关系不一致会有问题，会造成rebalance异常，导致队列无消费者消费

**Comment by itming001** [comment]
你这样配置消息消费成功之后 消息还会再次投递吗 我现在碰到第一次消费成功  后续投递了很多次
